ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Overview/useNodeCves.ts

```diff
@@ -0,0 +1,79 @@
+import { gql, useQuery } from '@apollo/client';
+import { QuerySearchFilter } from '../../types';
+import { getRegexScopedQueryString } from '../../utils/searchUtils';
+
+const cvesListQuery = gql`
+    query getNodeCVEs($query: String, $pagination: Pagination) {
+        nodeCVEs(query: $query, pagination: $pagination) {
+            cve
+            nodeCountBySeverity {
+                critical {
+                    total
+                }
+                important {
+                    total
+                }
+                moderate {
+                    total
+                }
+                low {
+                    total
+                }
+            }
+            topCVSS
+            affectedNodeCount
+            firstDiscoveredInSystem
+            distroTuples {
+                summary
+                operatingSystem
+                cvss
+                scoreVersion
+            }
+        }
+    }
+`;
+
+// TODO Need to verify these types with the BE implementation
+export type NodeCVE = {
+    cve: string;
+    nodeCountBySeverity: {
+        critical: { total: number };
+        important: { total: number };
+        moderate: { total: number };
+        low: { total: number };
+    };
+    topCVSS: number;
+    affectedNodeCount: number;
+    firstDiscoveredInSystem: string;
+    distroTuples: {
+        summary: string;
+        operatingSystem: string;
+        cvss: number;
+        scoreVersion: string;
+    }[];
+};
+
+export default function useNodeCves(
+    querySearchFilter: QuerySearchFilter,
+    page: number,
+    perPage: number
+) {
+    return useQuery<
+        { nodeCVEs: NodeCVE[] },
+        {
+            query: string;
+            pagination: {
+                offset: number;
+                limit: number;
+            };
+        }
+    >(cvesListQuery, {
+        variables: {
+            query: getRegexScopedQueryString(querySearchFilter),
+            pagination: {
+                offset: (page - 1) * perPage,
+                limit: perPage,
+            },
+        },
```

Comment:
An unfiltered idea for brainstorming: pro and con of helper function to compute pagination object, given
* `page`
* `perPage`

When my mind paused for a rest, my eyes fell on these lines, and I remembered my math mistake.

Reply:
I had a similar idea, and if I recall there have been a couple of issues lately where not using a helper function has led to a bug in pagination behavior.

Since this is a refactoring PR, I'll see what the impact is of adding such a function here.